src/Registry/TypeRegistry.php

```diff
@@ -174,7 +174,12 @@ public function __construct() {
 	 *
 	 * @return string
 	 */
-	protected function format_key( $key ) {
+	protected function format_key( string $key ) {
+
+		if ( false === strpos( $key, 'graphql_type_', 0 ) ) {
+			return strtolower( 'graphql_type_' . $key );
+		}
+
 		return strtolower( $key );
```

Comment:
We should just always prepend. Otherwise we would create a collision between two user-defined types named `header` and `graphql_type_header`. Maybe unlikely, but not much to gain with this logic on an internal-facing key.
```suggestion
		return strtolower( 'graphql_type_' . $key );
```

Reply:
I think the problem with _always_ prepending is that the format_key gets used on inserting types into the registry _and_ getting them out. 

If we always prepend and try: `get_type( 'graphql_type_post' )` then it will actually try to do `get_type( 'graphql_type_graphql_type_post' )` and find nothing. 

I guess I was trying to guard agains explicitly asking for `graphql_type_$name`, but I think you're right that it should be safe to always prepend 🤔

Reply:
I think all this key logic needs to be localized in `has_type` and `get_type` (as mentioned below) and then it should work.